packages/xstate-inspect/src/browser.ts

```diff
@@ -175,13 +180,14 @@ export function inspect(
     };
 
     service.subscribe((state) => {
-      // filter out synchronous notification from within `.start()` call when the `service.state` has not yet been assigned
+      // filter out synchronous notification from within `.start()` call
+      // when the `service.state` has not yet been assigned
       if (state === undefined) {
         return;
       }
       inspectService.send({
         type: 'service.state',
-        state: stringify(state),
+        state: stringifyState(state, options?.serialize),
```

Comment:
Note for a future improvement - when sending things through `postMessage` the browser uses "structured clone algorithm" (a browser API to use this algorithm, `structuredClone`, is becoming standardized lately) and this algorithm handles cyclic structures just fine. By dumping this to a string we lose information here - and maybe there would be a way to not do that in the future.

Reply:
```suggestion
        // TODO: investigate usage of structuredClone in browsers if available
        state: stringifyState(state, options?.serialize),
```